pkg/util/attributes/parser.go

```diff
@@ -48,28 +48,35 @@ func ParseV2Attributes(attrs []string, excl []string) ([]*netmap.NodeAttribute,
 			key := pair[0]
 			value := pair[1]
 
-			if at, ok := cache[key]; ok && at.Value() != value {
+			attribute, present := cache[key]
+			if present && attribute.Value() != value {
 				return nil, errNonUniqueBucket
 			}
 
 			if _, ok := restricted[key]; ok {
 				return nil, errUnexpectedKey
 			}
 
-			// replace non-printable symbols with escaped symbols without escape character
-			key = replaceEscaping(key, true)
-			value = replaceEscaping(value, true)
+			if !present {
+				// replace non-printable symbols with escaped symbols without escape character
+				key = replaceEscaping(key, true)
+				value = replaceEscaping(value, true)
 
-			attribute := netmap.NewNodeAttribute()
-			attribute.SetKey(key)
-			attribute.SetValue(value)
+				attribute = netmap.NewNodeAttribute()
+				attribute.SetKey(key)
+				attribute.SetValue(value)
+
+				cache[key] = attribute
```

Comment:
Can `key` be changed by `replaceEscaping` above? If so, we have different key formats when read and write the cache, is it ok?

Reply:
It is definitely not okay. I will add a test for that and will fix it.

Reply:
Done in https://github.com/nspcc-dev/neofs-node/pull/787/commits/45f40bea1140819d90d9ef12e0339e3af4659c40